core/shared/src/main/scala/zio/metrics/MetricClient.scala

```diff
@@ -43,4 +44,13 @@ private[zio] object MetricClient {
    */
   final def snapshot()(implicit unsafe: Unsafe): Set[MetricPair.Untyped] =
     metricRegistry.snapshot()
+
+  trait Listener {
```

Comment:
For performance reasons, I would create `Listener.Unsafe`, which is an unsafe version of this trait. Then `Listener.Unsafe` can extend `Listener`. This will allow us to create an even lower-level interface to listening to changes.

Reply:
Yeah, that makes sense. I changed the Listener methods to unsafe (returning `Unit`) for performance reasons, but I dropped the high-level interface as I don't think the use case for Datadog distribution would cover that.